@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -731,30 +731,54 @@ export class TileGeometryCreator {
                     object.userData.geometryId = srcGeometry.uuid;
                 }
 
-                if (isFillTechnique(technique) && (technique.clipping || technique.transparent)) {
-                    object.onBeforeRender = (_renderer, _scene, _camera, _geometry, _material) => {
-                        if (_material.clippingPlanes === null) {
-                            _material.clippingPlanes = this.clippingPlanes;
-                            // TODO: Add clipping for Spherical projection.
-                        } else if (mapView.projection.type === ProjectionType.Planar) {
-                            const planes = _material.clippingPlanes;
-                            const rightConstant =
-                                tile.center.x - mapView.worldCenter.x + tile.boundingBox.extents.x;
-                            planes[0].constant = rightConstant;
-
-                            const leftConstant =
-                                tile.center.x - mapView.worldCenter.x - tile.boundingBox.extents.x;
-                            planes[1].constant = -leftConstant;
-
-                            const topConstant =
-                                tile.center.y - mapView.worldCenter.y + tile.boundingBox.extents.y;
-                            planes[2].constant = topConstant;
-
-                            const bottomConstant =
-                                tile.center.y - mapView.worldCenter.y - tile.boundingBox.extents.y;
-                            planes[3].constant = -bottomConstant;
+                if (isFillTechnique(technique)) {
+                    object.onBeforeRender = chainCallbacks(
+                        object.onBeforeRender,
+                        (_renderer, _scene, _camera, _geometry, _material) => {
```

Comment:
yes, now it is correct and also very expensive we do a function call for each filled polygon :)